### DISCUSSION TITLE
Programmatically register excluded service definition / modify existing service definition

### DISCUSSION THREAD

**Discussion by SteNafH** [symfony/symfony#54591] [Q&A]
Previously, prior to Symfony 6.2, it was possible to dynamically register classes from the user's src directory within a Symfony Bundle. This was achieved by leveraging the applications `services.yaml` file to exclude certain classes from being loaded. However, changes introduced in Symfony 6.2 have rendered this approach ineffective.

Previous Behavior (Pre-Symfony 6.2):
In Symfony versions before 6.2, registering classes within a Symfony Bundle could be accomplished as follows:

```
class MyExtension extends Extension
{
    public function load(array $configs, ContainerBuilder $container): void
    {
        $classes = ['App\Handlers\UserClass'];

        foreach ($classes as $class) {
            $container->register($class)
                ->setAutowired(true)
                ->addTag('messenger.message_handler', ['bus' => 'my.bus']);
        }
    }
}
```

And in the applications `services.yaml` say
```
  App\:
    resource: '../src/'
    exclude:
      - '../src/Kernel.php'
      - '../src/Handlers/*'
```
 
This would then result in the Class to be registered as definition in the ContainerBuilder.

Since Symfony 6.2 this is not possible anymore due to this change:
https://github.com/symfony/symfony/pull/46279

In the `Symfony/Component/DependencyInjection/Loader/FileLoader.php/LoaderFileLoader.php` the following was added...
https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/DependencyInjection/Loader/FileLoader.php#L251

This results in classes to be added to the definitions even if the resource is excluded. 

My Bundle will register the services correctly still, but at 
https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php#L91

It gets overwritten again...

Is it not possible to do this anymore? the only workaround I have found is not mentioning the class at all in the `services.yaml` so it doesn't get added as a "container.excluded" definition.

Basically I need to overwrite an existing excluded definition or I need to modify an existing definition. How can I accomplish this?
